### PR TITLE
cleanup the camel-jms codebase a bit

### PR DESCRIPTION
**Pull request by bvahdat**

### CODE REVIEW

File: components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsMessageBodySetNullTest.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public void testSetNullBodyUsingSetBody() throws Exception {
             public void configure() throws Exception {
                 from("jms:queue:foo")
                         .to("mock:foo")
-                        .setBody(constant(null))
+                        .setBody(constant(new Object[] { null }))
```
Comment (davsclaus) [inline components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsMessageBodySetNullTest.java:86]: This looks weird/wrong. Setting null as before can be how end users would set a null body.

Comment (bvahdat) [reply components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsMessageBodySetNullTest.java:86]: Without that change the compiler would complain as:

```
[WARNING] /Users/bvahdat/dev/workspace/camel/components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsMessageBodySetNullTest.java:[86,42] non-varargs call of varargs method with inexact argument type for last parameter;
```

The problem is that `BuilderSupport#constant` is overloaded (which doesn't really fits for the `null` case):

```
public ValueBuilder constant(Object value)

public ValueBuilder constant(Object... value)
```

And the compiler thinks we intend to invoke the second one and gives that warning above. To overcome that I did that change to make it clear and resolve that warning.

Instead we could also do:

```
.setBody(simple("${null}"))
```

Comment (bvahdat) [reply components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsMessageBodySetNullTest.java:86]:
```
.setBody(constant((Object) null))
```

Would be the other option.

Comment (davsclaus) [reply components/camel-jms/src/test/java/org/apache/camel/component/jms/JmsMessageBodySetNullTest.java:86]: Ah okay then simple seems as the best

**Review by davsclaus — APPROVED** [review]

**Pull request merged**